dashboard/src/components/AppView/AppControls/AppControls.tsx

```diff
@@ -44,12 +45,18 @@ class AppControls extends React.Component<IAppControlsProps, IAppControlsState>
       <div className="AppControls">
         {/* If the app has been deleted hide the upgrade button */}
         {!deleted && (
-          <UpgradeButton
-            newVersion={app.updateInfo && !app.updateInfo.upToDate}
-            releaseName={name}
-            releaseNamespace={namespace}
-            push={push}
-          />
+          <React.Fragment>
+            <UpgradeButton
+              newVersion={app.updateInfo && !app.updateInfo.upToDate}
+              releaseName={name}
+              releaseNamespace={namespace}
+              push={push}
+            />
+            {/* We only show the rollback button if there are versions to rollback */}
+            {app.version > 1 && (
+              <RollbackButtonContainer releaseName={name} namespace={namespace} />
```

Comment:
I believe we're not using `React.Suspense` and `lazy` in this project yet, but this is a good component to be loaded lazily in this view.

Reply:
That's an interesting feature, didn't know about it. As far as I understand, this is to avoid rendering heavy components that will be used just in certain case. If I use `lazy` with this `RollbackButtonContainer` it will be always rendered isn't it? What would be the difference between using it or not?

Reply:
Actually, it's much powerful than that :). `React.lazy` allows you to download a component lazily when the user access to the view. This approach introduces two improvements in the applications:

* The component is not added to the `main` bundle of the application. As you mentioned, this is super useful for big components that are used only in a single view
* Since you can lazy load secondary components, the application performance increases due to the reduction of components in the view and the size of the bundle

`React.Suspense` is a component that controls lazy components in the UI. Any time you use `React.lazy` to load a component, it should be added as a child of a `React.Suspense` component. Suspense receives a `fallback` component as a prop that will be displayed while the lazy component is loaded. This `fallback` could be a text, a loader component or `null` if you don't need to display anything.

Another good feature about Suspense is that you can have several components in the component hierarchy. The closest one to the lazy component is the one that will render the fallback. It means you don't need to show a loader in the entire view if you're loading a lazy component.

> If I use lazy with this RollbackButtonContainer it will be always rendered isn't it? What would be the difference between using it or not?

About the example you mentioned, `RollbackButtonContainer` won't be rendered at the beginning. If you scope this component inside a `React.Suspense` component, the `fallback` will be returned. In this case, you can even return `null` and don't display anything. 

Once the component is fully loaded, it will be mounted in the UI.

Reply:
I am actually undoing this change because I am not able to test this if I change it for a suspense/lazy block.